### DISCUSSION TITLE
Lesson 7 : Getting hard to undestand and learn in end of lesson 7

### DISCUSSION THREAD

**Discussion by cell0901** [Cyfrin/foundry-full-course-cu#1462] [General]
At the end of lesson 7 when Patrick brings more tests , scripts and interactions. i was having hard time understanding this although it is doable but quite tricky for me. i have already gone through the course in November and faced similar issues at this phase but i skipped some months of practicing. Any tips or guidance?

**Comment by EngrPips** [comment]
Hello @cell0901 , I use to have similar problem as yours with all of this programming stuffs until few months ago when i realize that at the core of everything that exist there is the design and the purpose. You see understanding the purpose is the simplest part but yet the fundamental part to understand so as to understand everything about that thing. Let's take a car for example, what is the purpose of a car ? I would say it is transporting human and goods from one point to another. Once you understand this sole purpose of a car then it would make sense to you why it has an engine, why it has a tyre , why it has seat in it and so on. 

So all i am trying to say is whenever patrick wants to write a test, script, or an interactions file , make sure to ask yourself what is the purpose of this test , script, or interactions file. Once you are able to know what the purpose is the design would be making sense to you gradually till you now fully understand why and how he does everything he does. This is the way i have been approaching all my learning for the past few months and it has really make a better person at understanding stuffs.

My main point is, try to understand the `why` first before learning the `how`. because with a clear `why` a complicated `how` becomes easy
